scripts/decompose_flow_vectors

```diff
@@ -14,38 +14,102 @@ import sys
+class InvalidSourceError(Exception):
+    pass
+
+
+class NoDimensionsError(InvalidSourceError):
+    pass
+
+
+class NoFlowVariablesError(InvalidSourceError):
+    pass
+
+
+# VariableNotFoundError, NotAFlowVariableError and NotAVicModelError inherit InvalidVariableError
+class InvalidVariableError(Exception):
+    pass
+
+
+class VariableNotFoundError(InvalidVariableError):
+    pass
+
+
+class NotAFlowVariableError(InvalidVariableError):
+    pass
+
+
+class NotAVicModelError(InvalidVariableError):
+    pass
 
-def main(args):
-    # check that source file is usable:
-    source = Dataset(args.source_file, "r", format="NETCDF4")
+
+def source_check(source):
+    source_file_path = source.filepath()
 
     if not "lon" in source.dimensions or not "lat" in source.dimensions:
-        logger.debug(
+        logger.critical(
             "{} does not have latitude and longitude dimensions".format(
-                args.source_file
+                source_file_path
             )
         )
+        raise NoDimensionsError
+
+    valid_variables = []
+    for v in source.variables:
+        variable = source.variables[v]
+        if (
+            hasattr(variable, "dimensions")
+            and "lon" in variable.dimensions
+            and "lat" in variable.dimensions
+        ):
+            if np.ma.max(variable[:]) <= 9 and np.ma.min(variable[:]) >= 1:
+                valid_variables.append(v)
+
+    if len(valid_variables) == 0:
+        logger.critical(
+            "{} does not have a valid flow variable".format(source_file_path)
+        )
         source.close()
-        sys.exit()
+        raise NoFlowVariablesError
 
-    if not args.variable in source.variables:
-        logger.debug(
-            "Variable {} is not found in {}".format(args.variable, args.source_file)
+
+def variable_check(source, variable):
+    source_file_path = source.filepath()
+
+    if not variable in source.variables:
+        logger.critical(
+            "Variable {} is not found in {}".format(variable, source_file_path)
         )
         source.close()
-        sys.exit()
+        raise VariableNotFoundError
 
-    flow_variable = source.variables[args.variable]
+    flow_variable = source.variables[variable]
 
     if not "lon" in flow_variable.dimensions or not "lat" in flow_variable.dimensions:
-        logger.debug("Variable {} is not associated with a grid".format(args.variable))
+        logger.critical("Variable {} is not associated with a grid".format(variable))
         source.close()
-        sys.exit()
+        raise NotAFlowVariableError
 
     if np.ma.max(flow_variable[:]) > 9 or np.ma.min(flow_variable[:]) < 1:
-        logger.debug("Variable {} is not a valid flow routing".format(args.variable))
+        logger.critical("Variable {} is not a valid flow routing".format(variable))
         source.close()
-        sys.exit()
+        raise NotAVicModelError
+
+
+def main(args):
```

Comment:
Two things here:
1. We probably want to change the name of the main method to something more descriptive since we will be using it in another repo.
2. Knowing we will be using this in `thunderbird` we probably want to decompose the `args` param into individual components.

Reply:
I think we could maintain the `main()` function as it is now. The functions that `wps_decompose_flow_vectors` requires are `source_check`, `variable_check` and `dp.decompose_flow_vectors`, but `main()` won't be included. This `main()` function is specifically for `ce-dataprep` as far as I know. Please correct me if I am wrong.

Reply:
Ah ok sorry, I misread and thought this was in `dp/` and not `scripts/`.  That being said, if we want this to be importable to `thunderbird` these methods should be in `dp/decompose_vector_flows`.

Reply:
I agree with @nikola-rados . The `scripts` code should be minimal, just arranging arguments and calling the function(s) in `dp/`.